lib/RenderApp.pm

```diff
@@ -32,18 +32,31 @@ use RenderApp::Model::Problem;
 use RenderApp::Controller::RenderProblem;
 use RenderApp::Controller::IO;
 
+
 sub startup {
   my $self = shift;
   my $staticPath = $WeBWorK::Constants::WEBWORK_DIRECTORY."/htdocs/"; #curfile->dirname->sibling('public')->to_string.'/';
 
-  # config
-  $self->plugin('Config');
-  $self->plugin('TagHelpers');
-  $self->secrets($self->config('secrets'));
-  $ENV{problemJWTsecret} = $self->config('problemJWTsecret');
-  $ENV{webworkJWTsecret} = $self->config('webworkJWTsecret');
-  $ENV{SITE_HOST} = $self->config('SITE_HOST');
-  $ENV{JWTanswerURL} = $self->config('JWTanswerURL');
+
+	# Merge environment variables with config file
+	$self->plugin('Config');
+	$self->plugin('TagHelpers');
+	$self->secrets($self->config('secrets'));
+	$ENV{problemJWTsecret} //= $self->config('problemJWTsecret');
+	$ENV{webworkJWTsecret} //= $self->config('webworkJWTsecret');
+	$ENV{baseURL} //= $self->config('baseURL');
+	$ENV{formURL} //= $self->config('formURL');
+	$ENV{SITE_HOST} //= $self->config('SITE_HOST');
+	$ENV{JWTanswerURL} //= $self->config('JWTanswerURL');
+
+	# validate configration urls
+	if($ENV{baseURL} eq '/'){
+		$ENV{baseURL} = '';
+	}
+	#TODO
+	$ENV{baseURL} = $ENV{SITE_HOST}.$ENV{baseURL};
+	$ENV{formURL} = $ENV{baseURL}.$ENV{formURL};
```

Comment:
I don't know how to do this in Perl, but there should be a check like this:
```javascript
if (!$ENV{baseURL}.startsWith('http'))
     $ENV{baseURL} = $ENV{SITE_HOST}.$ENV{baseURL};
if (!$ENV{formURL}.startsWith('http'))
     $ENV{formURL} = $ENV{baseURL}.$ENV{formURL};
```

This is so that if baseURL and formURL are passed in as relative paths, this then constructs the absolute path.

Reply:
```perl
if ( $ENV{baseURL} !~ m|^https?://| ) {
     $ENV{baseURL} = $ENV{SITE_HOST}.$ENV{baseURL};
}
```
perl uses `=~` /regex/ and its negation `!~`; it also allows for alternate regex delimiters if prefaced with `m`. I have used them so that I don't need to escape the literal `/`s in the regex pattern.

Reply:
Thanks! I've implemented and tested those. Also, I've added a check to strip a trailing slash from SITE_HOST if present.